pkg/cloud/cloud.go

```diff
@@ -498,6 +498,104 @@ func (c *cloud) ModifyDisk(ctx context.Context, volumeID string, options *Modify
 	return c.waitForVolumeSize(ctx, volumeID)
 }
 
+func (c *cloud) ResizeOrModifyDisk(ctx context.Context, volumeID string, newSizeBytes int64, options *ModifyDiskOptions) (int64, error) {
+	if newSizeBytes != 0 {
+		klog.V(4).InfoS("Received Resize and/or Modify Disk request", "volumeID", volumeID, "newSizeBytes", newSizeBytes, "options", options)
+	} else {
+		klog.V(4).InfoS("Received Modify Disk request", "volumeID", volumeID,  "options", options)
+	}
+	
+	newSizeGiB := util.RoundUpGiB(newSizeBytes)
+	var oldSizeGiB int64
+	if newSizeBytes != 0 {
+		request := &ec2.DescribeVolumesInput{
+			VolumeIds: []*string{
+				aws.String(volumeID),
+			},
+		}
+		volume, err := c.getVolume(ctx, request)
+		if err != nil {
+			return 0, err
+		}
+
+		// AWS resizes in chunks of GiB (not GB)
```

Comment:
That's true. But it's distracting to mention it here. And anyway, the self-documenting line 508, has already made this point.